### ISSUE TITLE
Got undefined rows

### ISSUE THREAD

**Issue by subzeroDevelopment** [react-bootstrap-table/react-bootstrap-table2#505]
I have a BootstrapTable with SearchBar also a onSelectAll function, all works fine when select or unselect all rows without a previous search, but I am getting undefined rows when select  or unselect rows with filtered rows by SearchBar.
```jsx
<ToolkitProvider
                    keyField="id"
                    data={ this.props.data }
                    columns={ this.state.columns }
                    search
                >
                    {
                        props => (
                        <div>
                            <h3></h3>
                            <SearchBar { ...props.searchProps } 
                                placeholder="Buscar..." />
                            <hr />
                            <BootstrapTable
                                { ...props.baseProps } 
                                selectRow ={selectRow}
                                noDataIndication={'No hay comercios seleccionados'}
                                
                            />        
                        </div>
                        )
                    }
                </ToolkitProvider>
```
My selectRow Object:

```javascript
const selectRow = {
            mode: 'checkbox',
            clickToSelect: true,
            selected:this.state.selected,
            onSelect:this.handleOnSelect,
            onSelectAll:this.handleOnSelectAll
            
        }
```
handleOnSelectAll function only print rows now, here I'm getting undefined rows

```javascript
handleOnSelectAll = (isSelect, rows) => {
        console.log("rows",rows);
    }
```
this is the output of print in function:
rows (4) […]​0: undefined​1: undefined​2: undefined​3: undefined​length: 4​<prototype>: Array [] StoresDetail.js:64

**Comment by subzeroDevelopment** [comment]
I am using this versions:
 "react-bootstrap-table-next": "^1.1.1",   
 "react-bootstrap-table2-toolkit": "^1.0.1"

**Comment by AllenFang** [comment]
@subzeroDevelopment ok, I will check this out, seems like a bug lol

**Comment by AllenFang** [comment]
@subzeroDevelopment 

I can not reproduce this issue, following is my reproduce code:

```js
import BootstrapTable from 'react-bootstrap-table-next';
import ToolkitProvider, { Search } from 'react-bootstrap-table2-toolkit';

const { SearchBar } = Search;
const columns = [{
  dataField: 'id',
  text: 'Product ID'
}, {
  dataField: 'name',
  text: 'Product Name'
}, {
  dataField: 'price',
  text: 'Product Price'
}];

const selectRow = {
  mode: 'checkbox',
  clickToSelect: true,
  onSelect: (row, isSelect, rowIndex, e) => {
    console.log(row.id);
    console.log(isSelect);
    console.log(rowIndex);
    console.log(e);
  },
  onSelectAll: (isSelect, rows, e) => {
    console.log(isSelect);
    console.log(rows);
    console.log(e);
  }
};

<ToolkitProvider
  keyField="id"
  data={ products }
  columns={ columns }
  search
>
  {
    props => (
      <div>
        <h3>Input something at below input field:</h3>
        <SearchBar { ...props.searchProps } />
        <hr />
        <BootstrapTable
          { ...props.baseProps }
          selectRow={ selectRow }
        />
      </div>
    )
  }
</ToolkitProvider>
```

Did I missing anything? However, I saw you have a controlled selected state in `selectRow`, could you please share more about it? thanks

**Comment by AllenFang** [comment]
Reproduce steps:

1. search table with a text
2. select or select all
3. observer the output from `onSelect` and `onSelectAll`.

**Comment by AllenFang** [comment]
@subzeroDevelopment sorry, I can reproduce it. the problem is the selected item is not appear in search result then you trigger select all, the `onSelectAll` function still return the previous selected item even it is not in the current search result.  I will fix it in next release.

**Comment by subzeroDevelopment** [comment]
oh thanks, and regards.

**Comment by AllenFang** [comment]
@subzeroDevelopment fixed in `react-bootstrap-table-next@1.1.3`

https://react-bootstrap-table.github.io/react-bootstrap-table2/blog/2018/08/27/version-bump.html

Please reopen this issue and tag me if the problem still remain in newest version. thanks
